aws/resource_aws_appstream_fleet_test.go

```diff
@@ -0,0 +1,387 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/naming"
+)
+
+func testAccAwsAppStreamFleet_basic(t *testing.T) {
+	var fleetOutput appstream.Fleet
+	resourceName := "aws_appstream_fleet.test"
+	instanceType := "stream.standard.small"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckHasIAMRole(t, "AmazonAppStreamServiceAccess")
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAwsAppStreamFleetDestroy,
+		ErrorCheck:        testAccErrorCheck(t, appstream.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsAppStreamFleetConfigNameGenerated(instanceType),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppStreamFleetExists(resourceName, &fleetOutput),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "state", appstream.FleetStateRunning),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_time"),
+				),
+			},
+			{
+				Config:            testAccAwsAppStreamFleetConfigNameGenerated(instanceType),
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccAwsAppStreamFleet_Name_Generated(t *testing.T) {
```

Comment:
```suggestion
func testAccAwsAppStreamFleet_generatedName(t *testing.T) {
```

The last chunk of name is the specific test and should be _lowercase_. The second to last chunk is a subcategory and should be _uppercase_. (`testAccAws<Service>_<Subcategory>_<specificTest>`)

However, with only 6 tests, there's no need to use subcategories.